### PR TITLE
Feature/229 prepend version to GitHub release

### LINKED ISSUE #229: Prepend version (tag) to Github release title
## Background

Our internal policy states that a Github release title should start with the version number.
Thus, it would be nice if release-droid already could create the release title of Github releases in the correct format. Currently, release-droid just copies the code name found in the changelog.
It should be technically simple to prepend the version to the title, as the version is already known (release-droid knows which tag to create).

## Acceptance Criteria

New Github release drafts, created by release-droid, already should have the version as prefix in the release title.

### PR DESCRIPTION
**Pull request by ckunki**
fixes #229

### CODE REVIEW

File: src/main/java/com/exasol/releasedroid/adapter/github/GitHubReleaseMaker.java [hunk 1]
```diff
@@ -48,8 +48,8 @@ private String createGitHubRelease(final Repository repository) {
 
     private GitHubRelease createReleaseModel(final Repository repository, final String version) {
         final ReleaseLetter releaseLetter = repository.getReleaseLetter(version);
+        final String header = version + releaseLetter.getHeader().map(h -> ": " + h).orElse("");
```
Comment (tkilias) [inline src/main/java/com/exasol/releasedroid/adapter/github/GitHubReleaseMaker.java:51]: General question, should not have the release letter have always a header? If so, we should not attach the empty string, if it is not available, but instead fail.

Comment (ckunki) [reply src/main/java/com/exasol/releasedroid/adapter/github/GitHubReleaseMaker.java:51]: Hi @tkilias I will check.

Comment (ckunki) [reply src/main/java/com/exasol/releasedroid/adapter/github/GitHubReleaseMaker.java:51]: I can understand your request and share your opinion. 
[design.md](https://github.com/exasol/release-droid/blob/main/doc/design.md) only says 

> RD extracts the GitHub release's title from the release letter.

Do we need to update or detail the requirements first?

Comment (ckunki) [reply src/main/java/com/exasol/releasedroid/adapter/github/GitHubReleaseMaker.java:51]: @redcatbear : 
> Codename is  mandatory.

File: doc/design.md [hunk 2]
```diff
@@ -398,7 +398,7 @@ Needs: impl
 #### Retrieve GitHub Release Header from Release Letter
 `dsn~retrieve-github-release-header-from-release-letter~1`
```
Comment (redcatbear) [inline doc/design.md:399]: You need to update the revision number of the requirement if you change the semantics.

Comment (redcatbear) [inline doc/design.md:399]:
```suggestion
`dsn~retrieve-github-release-header-from-release-letter~2`
```
And update the dependent requirements of course.

**Review by redcatbear — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/releasedroid/adapter/github/GitHubReleaseMaker.java [hunk 3]
```diff
@@ -48,11 +49,19 @@ private String createGitHubRelease(final Repository repository) {
 
     private GitHubRelease createReleaseModel(final Repository repository, final String version) {
         final ReleaseLetter releaseLetter = repository.getReleaseLetter(version);
+        Optional<String> header = releaseLetter.getHeader();
+        if (header.isEmpty()) {
+            throw new IllegalStateException("Release header must not be empty.");
```
Comment (tkilias) [inline src/main/java/com/exasol/releasedroid/adapter/github/GitHubReleaseMaker.java:54]: Do we already use Error Codes in this project?

**Review by tkilias — DISMISSED** [review]

File: doc/design.md (hunk 2 above)
Comment (ckunki) [reply doc/design.md:399]: wilco next commit

File: src/main/java/com/exasol/releasedroid/adapter/github/GitHubReleaseMaker.java (hunk 3 above)
Comment (ckunki) [reply src/main/java/com/exasol/releasedroid/adapter/github/GitHubReleaseMaker.java:54]: yes, I will allocate a new error code

File: pom.xml [hunk 4]
```diff
@@ -24,8 +26,6 @@
             <artifactId>jira-rest-java-client-core</artifactId>
             <version>5.2.4</version>
         </dependency>
-        <!-- Update spring-beans version required by jira client to fix security issues
-             CVE-2022-22965, CVE-2016-1000027, CVE-2021-22118, CVE-2021-22096, CVE-2022-22950, CVE-2022-22965 -->
```
Comment (kaklakariada) [inline pom.xml:28]: Is this not necessary any more?

File: src/main/java/com/exasol/releasedroid/adapter/github/GitHubReleaseMaker.java [hunk 5]
```diff
@@ -48,11 +49,22 @@ private String createGitHubRelease(final Repository repository) {
 
     private GitHubRelease createReleaseModel(final Repository repository, final String version) {
         final ReleaseLetter releaseLetter = repository.getReleaseLetter(version);
+        String header = releaseLetter.getHeader().orElse("");
+        if (header.isEmpty()) {
+            throw new IllegalStateException(ExaError.messageBuilder("E-RD-GH-28") //
+                    .message("Release header must not be empty.") //
+                    .mitigation("Please provide release letter with non-empty header.") //
+                    .toString());
+        }
```
Comment (kaklakariada) [inline src/main/java/com/exasol/releasedroid/adapter/github/GitHubReleaseMaker.java:58]: There is also a method `Optional.orElseThrow()`. But this also catches the case when the header is `""` 👍

**Review by kaklakariada — APPROVED** [review]

File: pom.xml (hunk 4 above)
Comment (ckunki) [reply pom.xml:28]: Hi  @kaklakariada -- this was fixed in issue #233 by updating spring-beans but forgot to remove comment at that time :)

**Review by redcatbear — APPROVED** [review]

File: pom.xml (hunk 4 above)
Comment (kaklakariada) [reply pom.xml:28]: 👍

**Pull request merged**